### PR TITLE
fix(sct_events): race condition in EventsDevice

### LINKED ISSUE #1879: Artifacts test missing a coredump happing during install
### Prerequisites

* [ ] Are you rebased to master ?
* [ ] Is it reproducible ?
* [ ] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: master
* **scylla**: master 

### Logs
* test_id: 4bb52f7c-f026-41bd-bc85-aa3d0d4171e2
* job log: https://jenkins.scylladb.com/view/master/job/scylla-master/job/artifacts/job/artifacts-centos8/3/

### Description

Seems like the Artifact test finish too fast, and doesn't wait for the coredump to finish uploading.

### PR DESCRIPTION
**Pull request by enaydanov**
Trello: https://trello.com/c/IHoBCFLo/1708-remove-race-condition-between-events-threads

Closes #1879 

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/sct_events.py [hunk 1]
```diff
@@ -72,6 +83,18 @@ def wait_till_event_loop_is_working(number_of_events):
             except TimeoutError:
                 raise RuntimeError("Event loop is not working properly")
 
+    @property
+    def sub_port(self):
+        if self._ready.wait():
```
Comment (bentsi) [inline sdcm/sct_events.py]: Let's add a timeout
```suggestion
        if self._ready.wait(timeout=30):
```

File: sdcm/sct_events.py [hunk 2]
```diff
@@ -72,6 +83,18 @@ def wait_till_event_loop_is_working(number_of_events):
...
+            return self._sub_port
+        raise RuntimeError("EventsDevice is not ready to send events.")
+
+    @property
+    def pub_port(self):
+        if self._ready.wait():
```
Comment (bentsi) [inline sdcm/sct_events.py]: same here

File: sdcm/sct_events.py (hunk 1 above)
Comment (enaydanov) [reply sdcm/sct_events.py]: done

File: sdcm/sct_events.py (hunk 2 above)
Comment (enaydanov) [reply sdcm/sct_events.py]: done

**Comment by bentsi** [comment]
@enaydanov do you think that it is critical to backport this PR? There are conflicts, that I am not sure how to resolve. So if you think it is critical let's create backport PR for branch-3.3 first.

**Comment by enaydanov** [comment]
> @enaydanov do you think that it is critical to backport this PR? There are conflicts, that I am not sure how to resolve. So if you think it is critical let's create backport PR for branch-3.3 first.

It will cause intermittent failures of artifacts jobs. So, I think, it's worth to be backported.

**Pull request merged**